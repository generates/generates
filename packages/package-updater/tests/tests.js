import { promises as fs } from 'fs'
import path from 'path'
import { test } from '@ianwalter/bff'
import { update } from '../index.js'

const cwd = 'tests/tmp'

test('updating package.json', async t => {
  const release = { logLevel: 'debug', registries: ['foo', 'bar'] }
  await update({ release }, { cwd })
  const packagePath = path.resolve('tests/tmp/package.json')
  t.expect(await fs.readFile(packagePath, 'utf8')).toMatchSnapshot()
})

test('updating package.json with existing data', async t => {
  const $package = { bff: { logLevel: 'debug' }, num: 8 }
  const bff = { logLevel: 'info', plugins: ['config.js'] }
  await update({ bff }, { cwd, package: $package })
  const packagePath = path.resolve('tests/tmp/package.json')
  t.expect(await fs.readFile(packagePath, 'utf8')).toMatchSnapshot()
})
