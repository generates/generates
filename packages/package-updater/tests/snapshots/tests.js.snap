// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updating package.json 1`] = `
"{
  \\"name\\": \\"@ianwalter/example\\",
  \\"release\\": {
    \\"registries\\": [
      \\"foo\\",
      \\"bar\\"
    ],
    \\"logLevel\\": \\"debug\\"
  }
}
"
`;

exports[`updating package.json with existing data 1`] = `
"{
  \\"bff\\": {
    \\"logLevel\\": \\"info\\",
    \\"plugins\\": [
      \\"config.js\\"
    ]
  },
  \\"num\\": 8
}
"
`;
